### PR TITLE
changed imageView to a custom version

### PR DESCRIPTION
**Pull request by AmitSilber**

### CODE REVIEW

File: app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java [hunk 1]
```diff
@@ -34,8 +46,11 @@
     private static final int MAX_RECIPES = 10;
     static final String INGREDIENTS = "Ingredients";
     static final String COOKING_TIME = "CookingTime";
+    static final Integer ITEM_PRESSED = 70;
+    static final Integer ITEM_UNPRESSED = 255;
 
     private IngredientsList ingredientsList;
+    private Integer[] COLOURS = new Integer[5];
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java]: probably should be `COLORS` or `COLORS_ARRAY` (without the u in colors)

File: app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java [hunk 2]
```diff
@@ -89,15 +110,17 @@ public void onItemClick(AdapterView<?> parent, View v,
                                     int position, long id) {
                 ImageAdapter adapter = (ImageAdapter) gridview.getAdapter();
                 adapter.setIsPressed(position);
+                ImageView imageView;
                 if (position < ImageAdapter.INGREDIENTS_AMOUNT) {
                     if (adapter.getIsPressed(position)) {
-                        ImageView imageView = (ImageView) v;
-                        imageView.setImageAlpha(100);
+//                        imageView = (ImageView) v.findViewById(R.id.custom_image);
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java]: can remove this line

File: app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java [hunk 3]
```diff
@@ -89,15 +110,17 @@ public void onItemClick(AdapterView<?> parent, View v,
...
+                        v.setBackgroundColor(COLOURS[position % 5]);
                         ingredientsList.addIngredient(adapter.getIngredientName(position));
 
                     } else {
-                        ImageView imageView = (ImageView) v;
-                        imageView.setImageAlpha(255);
+//                        imageView = (ImageView) v.findViewById(R.id.custom_image);
+//                        imageView.setImageAlpha(ITEM_UNPRESSED);
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java]: remove comments

File: app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java [hunk 4]
```diff
@@ -50,9 +52,40 @@
             new Ingredient(R.drawable.lemon, "lemon"),
             new Ingredient(R.drawable.ginger, "ginger"),
             new Ingredient(R.drawable.grapes, "grapes"),
-            new Ingredient(R.drawable.apple, "apple"),
+            new Ingredient(R.drawable.apple, "apple")
+    };
+    // references to our images
+    private static Integer[] ingredientId = {
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java]: `ingredientIds`

File: app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java [hunk 5]
```diff
@@ -50,9 +52,40 @@
...
+            R.drawable.onion, R.drawable.rice,
+            R.drawable.beef, R.drawable.pasta,
+            R.drawable.sausage, R.drawable.spinach,
+            R.drawable.tomato, R.drawable.potatoes,
+            R.drawable.bacon, R.drawable.bell_pepper,
+            R.drawable.cabbage, R.drawable.ginger,
+            R.drawable.leek
     };
 
+    private static String[] ingredientName = {
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java]: `ingredientNames`

File: app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java [hunk 6]
```diff
@@ -84,25 +118,32 @@ public long getItemId(int position) {
...
+
+            customView = inflater.inflate(R.layout.imagetext_layout, null);
+            TextView customText = (TextView) customView.findViewById(R.id.custom_text);
+            ImageView customImage = (ImageView) customView.findViewById(R.id.custom_image);
+            customText.setText(ingredientName[position]);
+            customImage.setImageResource(ingredientId[position]);
+
         } else {
-            imageView = (ImageView) convertView;
+            customView = (View) convertView; // when does this happen?
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java]: i wrote the "// when does this happen?" :) you can remove it

File: app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java [hunk 7]
```diff
@@ -84,25 +118,32 @@ public long getItemId(int position) {
...
     }
     // create a new ImageView for each item referenced by the Adapter
 
     public View getView(int position, View convertView, ViewGroup parent) {
-        ImageView imageView;
+        View customView;
+        LayoutInflater inflater = (LayoutInflater) mContext.
+                getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+
         if (convertView == null) {
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java]: i know it's like this in the guide, but maybe it'll be better to switch the cases:
`if (convertView != null) {
return convertView 
} ... 
(else logic)`

**Review by mfocshaner — APPROVED** [review]
hey, great job! <3
I left some remarks but they are cosmetic changes, so you can fix them and merge.

File: app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java (hunk 1 above)
Comment (mharsat) [reply app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java]: "Colours" with a U is correct but British.

Comment (AmitSilber) [reply app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java]: I will end U

**Pull request merged**